android/src/main/java/com/swmansion/rnscreens/ScreenFragment.kt

```diff
@@ -169,29 +174,30 @@ open class ScreenFragment : Fragment {
     }
 
     private fun dispatchOnWillAppear() {
```

Comment:
I think all these event methods should be moved to `ScreenStackFragment` and I am not sure why are they even in this class at all. We only use them in `native-stack` since JS handles the lifecycle and animation progress events in JS stack.

Reply:
There is some logic that dispatches events in `ScreenFragment` (e.g.  in `onViewAnimationEnd()`) and uses it. I'll consider moving it in follow up refactors.